src/docbuilder/queue.rs

```diff
@@ -58,15 +58,20 @@ impl DocBuilder {
         let id: i32 = query.get(0).get(0);
         let name: String = query.get(0).get(1);
         let version: String = query.get(0).get(2);
+        let attempt: i32 = query.get(0).get(3);
 
         match builder.build_package(self, &name, &version) {
             Ok(_) => {
                 let _ = conn.execute("DELETE FROM queue WHERE id = $1", &[&id]);
             }
             Err(e) => {
                 // Increase attempt count
-                let _ = conn.execute("UPDATE queue SET attempt = attempt + 1 WHERE id = $1",
-                                     &[&id]);
+                let attempt = attempt + 1;
+                let _ = conn.execute("UPDATE queue SET attempt = $1 WHERE id = $2",
```

Comment:
To keep this atomic, you can consolidate the select and the update with PostgreSQL's [`RETURNING`](https://www.postgresql.org/docs/9.5/dml-returning.html):

```
UPDATE queue SET attempt + 1 WHERE id = $1 RETURNING attempt;
```